overlord/servicestate/quota_control_test.go

```diff
@@ -0,0 +1,322 @@
+		},
+		{
+			// called for the modified service unit files from EnsureSnapServices
+			// TODO: this call should go away?
+			expArgs: []string{"daemon-reload"},
+		},
+	})
+	defer r()
+
+	st := s.state
+	st.Lock()
+	// setup the snap so it exists
+	snapstate.Set(s.state, "test-snap", s.testSnapState)
+	snaptest.MockSnapCurrent(c, testYaml, s.testSnapSideInfo)
+	st.Unlock()
+
+	// trying to remove a group that does not exist fails
+	err := s.mgr.RemoveQuota("not-exists")
+	c.Assert(err, ErrorMatches, `cannot remove non-existent quota group "not-exists"`)
+
+	// create a quota
+	err = s.mgr.CreateQuota("foo", "", []string{"test-snap"}, quantity.SizeGiB)
+	c.Assert(err, IsNil)
+
+	// check that the quota groups was created in the state
+	st.Lock()
+	defer st.Unlock()
+	m, err := servicestate.AllQuotas(st)
+	c.Assert(err, IsNil)
+	c.Assert(m, HasLen, 1)
+	for name, grp := range m {
+		switch name {
+		case "foo":
+			c.Assert(grp.Snaps, DeepEquals, []string{"test-snap"})
+			c.Assert(grp.SubGroups, HasLen, 0)
+			c.Assert(grp.ParentGroup, Equals, "")
+		default:
+			c.Errorf("unexpected group %q in state", name)
+		}
+	}
+
+	// remove the quota from the state
+	st.Unlock()
+	defer st.Lock()
+	err = s.mgr.RemoveQuota("foo")
+	c.Assert(err, IsNil)
+
+	st.Lock()
+	defer st.Unlock()
+	m, err = servicestate.AllQuotas(st)
+	c.Assert(err, IsNil)
+	c.Assert(m, HasLen, 0)
+}
+
+func (s *quotaControlSuite) TestUpdateQuotaGroupNotExist(c *C) {
+	opts := servicestate.QuotaGroupUpdate{}
+	err := s.mgr.UpdateQuota("non-existing", opts)
+	c.Check(err, ErrorMatches, `group "non-existing" does not exist`)
+}
+
+func (s *quotaControlSuite) TestUpdateQuotaChangeMemLimit(c *C) {
+	r := s.mockSystemctlCalls(c, []expectedSystemctl{
+		{
+			// called for new slice unit written by CreateQuota after we create
+			// the snap in state
+			expArgs: []string{"daemon-reload"},
+		},
+		{
+			// called by UpdateQuota
+			expArgs: []string{"daemon-reload"},
+		},
+	})
+	defer r()
+
+	st := s.state
+	st.Lock()
+	// setup the snap so it exists
+	snapstate.Set(s.state, "test-snap", s.testSnapState)
+	snaptest.MockSnapCurrent(c, testYaml, s.testSnapSideInfo)
+	st.Unlock()
+
+	// create a quota group
+	err := s.mgr.CreateQuota("foo", "", []string{"test-snap"}, quantity.SizeGiB)
+	c.Assert(err, IsNil)
+
+	// ensure mem-limit is 1 GB
+	st.Lock()
+	m, err := servicestate.AllQuotas(st)
+	st.Unlock()
+	c.Check(err, IsNil)
+	c.Check(m, HasLen, 1)
+	c.Check(m["foo"].MemoryLimit, Equals, quantity.SizeGiB)
+
+	// modify to 2 GB
+	opts := servicestate.QuotaGroupUpdate{NewMemoryLimit: 2 * quantity.SizeGiB}
+	err = s.mgr.UpdateQuota("foo", opts)
+	c.Assert(err, IsNil)
+
+	// and check that it got updated in the state
+	st.Lock()
+	m, err = servicestate.AllQuotas(st)
+	st.Unlock()
+	c.Check(err, IsNil)
+	c.Check(m, HasLen, 1)
+	c.Check(m["foo"].MemoryLimit, Equals, 2*quantity.SizeGiB)
+
+	// XXX: should we look at the written snap services/slices here too?
```

Comment:
yes, it seems we could replace mostly ensureSnapServicesForGroup  with a daemon-reload and most tests here will still pass. It seems we should at least check in many of this tests for the existence of the .slice and whether .service files where changed

Reply:
done